src/test/java/com/citrix/elearning/candidatemerge/tests/CandidateMergeTest.java

```diff
@@ -15,65 +14,107 @@
-			System.out.println(masterDate);
+			logger.info("MasterDate in Date format=" + masterDate);
+
+			final boolean result = candidateReconciliationPage.isAlertPresent();
+			logger.info("Comparing emails and dates");
 			if (inboundEmailText.equals(masterEmailText) || inboundEmailText.equals(alternativeEmailText)) {
 				if (inboundDate.compareTo(masterDate) < 0 || inboundDate.compareTo(masterDate) == 0) {
 
 					final String dontApplyText = candidateReconciliationPage.getDontApplyText();
 					candidatePojo.setResult(dontApplyText);
+
+					logger.info("Clicking on Don't apply link");
 					candidateReconciliationPage.clickDontApply();
-					logger.info("Clicked on Don't apply");
-					candidateReconciliationPage.closeAlert();
+					if (result) {
+						final String applyLinkText = candidateReconciliationPage.alertText();
+						if (applyLinkText.equals(
+								"Please verify that you wish to update only the VUE record with the new data.")) {
+							candidateReconciliationPage.closeAlert();
+						} else {
+							candidateReconciliationPage.closeAlert();
+						}
+					}
+
 				} else if (inboundDate.compareTo(masterDate) > 0) {
+
 					final String applyText = candidateReconciliationPage.getApplyText();
 					candidatePojo.setResult(applyText);
+
+					logger.info("Clicking on apply link");
 					candidateReconciliationPage.clickApply();
-					logger.info("Clicked on apply");
-					candidateReconciliationPage.closeAlert();
-					System.out.println("popup closed");
+					if (result) {
+						final String applyLinkText = candidateReconciliationPage.alertText();
+						if (applyLinkText.equals(
+								"This inbound record has an older revision date than the Master Record.  The demographic information in the inbound record will not be applied to the Master Record.")) {
+							candidateReconciliationPage.closeAlert();
+						} else {
+							candidateReconciliationPage.closeAlert();
+						}
+					}
 				}
 			} else {
 				final String createNewText = candidateReconciliationPage.getCreateNewText();
 				candidatePojo.setResult(createNewText);
+
+				logger.info("Clicking on CreateNew link");
 				candidateReconciliationPage.clickCreateNew();
-				logger.info("Clicked on CreateNew");
-				candidateReconciliationPage.closeAlert();
+				if (result) {
+					final String createNewLinkText = candidateReconciliationPage.alertText();
+					if (createNewLinkText
+							.equals("Please verify that you wish to create a new candidate record with this data.")) {
+						candidateReconciliationPage.closeAlert();
+					} else {
+						candidateReconciliationPage.closeAlert();
+					}
+				}
 			}
-
+			logger.info("Going back to CandidateQueue by clicking CandidateQueue tab");
 			findCandidatePage.clickCandidateQueue();
+
+			logger.info("Searching by Name");
 			candidateQueuePage.searchTextboxAndSendName(numberParsed);
+
+			logger.info("By using combination of QueueDate and name,verifing whether record is there or not");
```

Comment:
Verify candidate record (candidate )is cleared from queue.
Assert.assertTrue/False();